### PR TITLE
finished broker utility implementation and testing

### PR DESCRIPTION
**Pull request by bar-jakubovitz**
[Finishes #94474624]
[Finishes #95607070]

### CODE REVIEW

**Comment by bar-jakubovitz** [comment]
@anatb Code review please.
finishes broker utility and testing
and minor fix to subscription api (added params on delete)

File: lib/nurego/cf/broker_utility.rb [hunk 1]
```diff
@@ -1,17 +1,24 @@
...
     class BrokerUtility
+
+      @external_ids = true
+
+      class << self
+        attr_accessor :external_ids
+      end
+
       def self.provision(params)
         # todo: what to return if nurego is notified ?
```
Comment (anatb) [inline lib/nurego/cf/broker_utility.rb]: remove the todo?

File: lib/nurego/cf/broker_utility.rb [hunk 2]
```diff
@@ -20,6 +27,8 @@ def self.update(params)
         raise InvalidRequestError.new('Invalid parameter instance_id', 'instance_id') unless params['instance_id']
         sub = Subscription.retrieve(params['instance_id'])
         sub.plan_id = params['plan_id']
+        sub.provider = 'cloud-foundry'
```
Comment (anatb) [inline lib/nurego/cf/broker_utility.rb]: maybe hold cloud-foundry as a constant somewhere...

File: lib/nurego/cf/broker_utility.rb [hunk 3]
```diff
@@ -36,65 +45,57 @@ def self.nurego_notified(params)
       end
 
       def self.get_service_catalog(service_id)
-        service = Service.retrieve()
+        service = Service.retrieve(service_id)
```
Comment (anatb) [inline lib/nurego/cf/broker_utility.rb]: if one wants a specific distribution channel - how will this be done? (I think this is a very near-future use case)

File: lib/nurego/subscription.rb [hunk 4]
```diff
@@ -24,8 +24,8 @@ def save
     end
 
     # Delete override
-    def delete
-      response, api_key = Nurego.request(:delete, url(self.organization_id), @api_key)
+    def delete(params={})
```
Comment (anatb) [inline lib/nurego/subscription.rb]: maybe put a comment explaining what params should hold for future generations?

File: spec/integration/broker_utility_spec.rb [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+describe "Broker Utility" do
+
+  before(:all) do
+    setup_nurego_lib
+    setup_login_and_login
+    Nurego::Cf::BrokerUtility.external_ids = false
+    @test_instance_id = SecureRandom.uuid
+  end
+
+  it "doesn't notifies on provision requests from proxy" do
```
Comment (anatb) [inline spec/integration/broker_utility_spec.rb]: notifies -> notify

**Comment by anatb** [comment]
excellent! (and you've also mastered the ruby code standards :+1: )

File: lib/nurego/cf/broker_utility.rb (hunk 3 above)
Comment (bar-jakubovitz) [inline lib/nurego/cf/broker_utility.rb]: Nurego::Cf::BrokerUtility.get_service_catalog({ id: SERVICE_ID, distribution_channel: 'cf' }).
its hidden as well in the service retrieve wrapper
Nurego::Service.retrieve({ id: SERVICE_ID, distribution_channel: 'website' })
for distribution channel

File: lib/nurego/subscription.rb (hunk 4 above)
Comment (bar-jakubovitz) [inline lib/nurego/subscription.rb]: used for passing extra parameters to nurego Bss (only use case now is the BrokerUtility that pass the skip_service_webhook=true and provider='cf'

**Pull request merged**